File: api/Service/LanguageService.cs
using Model;
using Persistence;
using System;
using System.Collections.Generic;
using System.Linq;

namespace Service
{
    public interface ILanguageService
    {
        IEnumerable<Language> GetAll();
        IEnumerable<Language> GetByIndex(int index);
        IEnumerable<Language> GetByName(string name);
        int GetCount();
        bool Add(Language model);
        bool Delete(int id);
        bool Update(Language model);
        Language Get(int id);
    }

    public class LanguageService:ILanguageService
    {
        private readonly LastaContext _LanguageDbContext;

        public LanguageService(
            LastaContext LanguageDbContext
        )
        {
            _LanguageDbContext = LanguageDbContext;
        }

        public IEnumerable<Language> GetAll()
        {
            var result = new List<Language>();

            try
            {
                result = _LanguageDbContext.Language.ToList();
            }
            catch (System.Exception)
            {

            }

            return result;
        }

        public IEnumerable<Language> GetByIndex(int index)
        {
            var result = new List<Language>();

            try
            {
                result = _LanguageDbContext.Language
                    .Skip(index * 10).Take(10)
                    .ToList();
            }
            catch (System.Exception)
            {

            }

            return result;
        }

        public IEnumerable<Language> GetByName(string name)
        {
            var result = new List<Language>();

            try
            {
                result = _LanguageDbContext
                           .Language
                         .Where(x => x.Name.ToLower().StartsWith(name.ToLower()))
                         .ToList();
            }
            catch (System.Exception)
            {

            }

            return result;
        }

        public bool Add(Language model)
        {
            try
            {
                _LanguageDbContext.Add(model);
                _LanguageDbContext.SaveChanges();
            }
            catch (System.Exception)
            {
                return false;
            }

            return true;
        }

        public bool Update(Language model)
        {
            Language OriginalModel = _LanguageDbContext.Language.Where(x => x.LanguageId == model.LanguageId).FirstOrDefault();
            OriginalModel.Name = model.Name;

            _LanguageDbContext.Language.Update(OriginalModel);
            _LanguageDbContext.SaveChanges();
            return true;
        }

        public Language Get(int id)
        {
            var result = new Language();

            try
            {
                result = _LanguageDbContext.Language.Single(x => x.LanguageId == id);

            }
            catch (System.Exception)
            {

            }

            return result;
        }

        public int GetCount()
        {
            decimal temp = 0;

            try
            {
                temp = _LanguageDbContext.Language.Count();
            }
            catch (System.Exception)
            {

            }
            temp /= 10;
            //Gets decimal of number - if number is 10.45 dec = 0.45
            var dec = temp - Math.Truncate(temp);

            int result = (int)temp;

            if (dec != 0)
                result++;

            return result;
        }

        public bool Delete(int id)
        {
            try
            {
                Language deleteModel = _LanguageDbContext.Language.Where(x => x.LanguageId == id).FirstOrDefault();
                if (deleteModel != null)
                    _LanguageDbContext.Language.Remove(deleteModel);
                else
                    return false;

            }catch(System.Exception ex)
            {
                return false;
            }
            return true;
        }
    }
}

File: client/src/services/LanguageService.js
function LanguageService(axios, baseUrl) {
  this.axios = axios;
  this.baseUrl = `${baseUrl}language`;

  this.get = function (id) {
    let self = this;
    return self.axios.get(`${self.baseUrl}/${id}`);
  }

  this.getAll = function () {
    let self = this;
    return self.axios.get(`${self.baseUrl}`);
  }

  this.getByIndex = function (index) {
    let self = this;
    return self.axios.get(`${self.baseUrl}/getByIndex/${index}`);
  }
  this.getByName = function (name) {
    let self = this;
    return self.axios.get(`${self.baseUrl}/getByName/${name}`);
  }
  this.getAccommodationCount = function () {
    let self = this;
    return self.axios.get(`${self.baseUrl}/getCount`);
  }
  this.getCount = function () {
    let self = this;
    return self.axios.get(`${self.baseUrl}/getCount`);
  }
  this.add = function (model) {
    let self = this;
    return self.axios.post(`${self.baseUrl}`, model);
  }

  this.update = function (model) {
    let self = this;
    return self.axios.put(`${self.baseUrl}`, model);
  }

  this.remove = function (id) {
    let self = this;
    return self.axios.delete(`${self.baseUrl}/${id}`);
  }
}

export default LanguageService